source/common/http/conn_manager_impl.cc

```diff
@@ -1092,6 +1136,31 @@ void ConnectionManagerImpl::ActiveStreamEncoderFilter::
 
 void ConnectionManagerImpl::ActiveStreamEncoderFilter::continueEncoding() { commonContinue(); }
 
+void ConnectionManagerImpl::ActiveStreamEncoderFilter::responseDataTooLarge() {
+  if (parent_.encoder_filters_all_streaming_) {
+    onEncoderFilterAboveWriteBufferHighWatermark();
+  } else {
+    // FIXME 500
+    HeaderMapPtr response_headers{new HeaderMapImpl{
```

Comment:
I think this case needs a bit of thinking. It is technically possible for the response to start, but then have buffering occur, so I think we need to check if response has started and if so reset, otherwise we can send a 500.